src/modules/launcher/PowerLauncher/Plugin/QueryBuilder.cs

```diff
@@ -12,58 +12,28 @@ namespace PowerLauncher.Plugin
 {
     public static class QueryBuilder
     {
-        public static Dictionary<PluginPair, Query> Build(ref string text)
+        public static Dictionary<PluginPair, Query> Build(string text)
         {
             if (text == null)
             {
                 throw new ArgumentNullException(nameof(text));
             }
 
-            // replace multiple white spaces with one white space
-            var terms = text.Split(new[] { Query.TermSeparator }, StringSplitOptions.RemoveEmptyEntries);
-            if (terms.Length == 0)
-            { // nothing was typed
-                return null;
-            }
+            text = text.Trim();
 
             // This Dictionary contains the corresponding query for each plugin
             Dictionary<PluginPair, Query> pluginQueryPair = new Dictionary<PluginPair, Query>();
-
-            var rawQuery = string.Join(Query.TermSeparator, terms);
-
-            // This is the query on removing extra spaces which would be executed by global Plugins
-            text = rawQuery;
-
-            string possibleActionKeyword = terms[0];
-
-            foreach (string pluginActionKeyword in PluginManager.NonGlobalPlugins.Keys)
+            foreach (var plugin in PluginManager.NonGlobalPlugins)
             {
-                // Using Ordinal since this is used internally
-                if (possibleActionKeyword.StartsWith(pluginActionKeyword, StringComparison.Ordinal))
+                var pluginActionKeyword = plugin.Metadata.ActionKeyword;
+                if (plugin.Metadata.Disabled || !text.StartsWith(pluginActionKeyword, StringComparison.Ordinal))
```

Comment:
The `Disabled` check is not needed here, and you also don't have it in the `foreach` on line 43. The check is done in `MainViewModel.cs`

Reply:
It would change the behavior of the function before we had this check. Maybe we can remove it.

Reply:
Removed `if`s with `Disabled` from `MainViewModel` and left it here as it does not make sense to return an entry if plugin is disabled.